### ISSUE TITLE
Getting file not found error while using bytepair embedding

### ISSUE THREAD

**Issue by Michael95-m** [flairNLP/flair#1643]
**Describe the bug**
@alanakbik , I got FileNotFoundError when I predicted with the model with character and  bytepair embedding. What I did was that I trained on a gpu and predicted on local cpu. Then it is showing file not found error with the gpu path. 

**To Reproduce**
```
from flair.data import Corpus
from flair.datasets import ColumnCorpus
from flair.embeddings import TokenEmbeddings, StackedEmbeddings, BertEmbeddings, CharacterEmbeddings, BytePairEmbeddings, WordEmbeddings
from flair.visual.training_curves import Plotter
from flair.models import SequenceTagger
from flair.trainers import ModelTrainer
from typing import List
from torch.optim.adam import Adam
import sys
import flair, torch 

flair.device = torch.device('cuda:0')

columns = {0: 'text', 1: 'ner'}
data_folder =  './data/Model_Trainer_Data'
corpus: Corpus = ColumnCorpus(data_folder, columns,train_file= sys.argv[1] , dev_file=sys.argv[3], test_file= sys.argv[4], in_memory=False)

tag_type = 'ner'
tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
# bert_embedding = BertEmbeddings('bert-base-multilingual-cased')

embedding_types: List[TokenEmbeddings] = [
    CharacterEmbeddings(),
    BytePairEmbeddings('my', 300),
    # bert_embedding
    ]

embeddings: StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                        embeddings=embeddings,
                                        tag_dictionary=tag_dictionary,
                                        tag_type=tag_type,
                                        rnn_layers = 1,
                                        use_crf=True)

# trainer: ModelTrainer = ModelTrainer(tagger, corpus, optimizer=Adam)
trainer: ModelTrainer = ModelTrainer(tagger, corpus)

trainer.train('./results/'+ sys.argv[2],
              learning_rate=0.1,
              embeddings_storage_mode = 'gpu',
              mini_batch_size=64,
              patience=5,
              max_epochs=150,
              train_with_dev=False,
              checkpoint=True)

# from pathlib import Path 

# checkpoint = 'results/model_chunk_6/checkpoint.pt'
# checkpoint = tagger.load_checkpoint(Path(checkpoint))
# trainer = ModelTrainer.load_from_checkpoint(checkpoint, corpus)

# trainer.train('./results/'+ sys.argv[2],
#               learning_rate=0.15,
#               embeddings_storage_mode = 'gpu',
#               mini_batch_size=16,
#               patience=3,
#               max_epochs=150,
#               train_with_dev=False,
#               checkpoint=True)


plotter = Plotter()
# plotter.plot_training_curves('./results/'+ sys.argv[2]+'/loss.tsv')
plotter.plot_weights('./results/'+ sys.argv[2]+'/weights.txt')


```

**Expected behavior**

I'd like to predict on cpu machines.

**Screenshots**

This is where I got error

![ss_1](https://user-images.githubusercontent.com/64765786/83096039-b47ab500-a0ca-11ea-91b5-bbdea6e9549c.png)

Showing to gpu path

![ss_2](https://user-images.githubusercontent.com/64765786/83096069-c9574880-a0ca-11ea-8715-9bbaa756837c.png)

**Environment (please complete the following information):**
 - OS [Linux-mint:19.3]:
 - Version [flair: 0.5 ]:

**Comment by Michael95-m** [comment]
@alanakbik  @djstrong , please help solve this issue. I'd like to use bytepair embedding with flair model in pos tagging model. But bytepair embedding got the error in flair 0.5 and I found that some model trained in flair 0.5 is not backward compatible with flair 0.4.5.

**Comment by alanakbik** [comment]
@Michael95-m this should now be fixed in master branch (see #1802) and will be part of the upcoming release
